DB/src/main/java/io/deephaven/db/v2/StreamTableToAppendOnlyTable.java

```diff
@@ -0,0 +1,82 @@
+package io.deephaven.db.v2;
+
+import io.deephaven.db.tables.Table;
+import io.deephaven.db.tables.utils.QueryPerformanceRecorder;
+import io.deephaven.db.v2.sources.ArrayBackedColumnSource;
+import io.deephaven.db.v2.sources.ColumnSource;
+import io.deephaven.db.v2.utils.ChunkUtils;
+import io.deephaven.db.v2.utils.Index;
+import io.deephaven.db.v2.utils.IndexShiftData;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * Converts a Stream Table to an in-memory append only table.
+ *
+ * Note, this table will grow without bound as new stream values are encountered.
+ */
+public class StreamTableToAppendOnlyTable {
+    /**
+     * Convert a Stream Table to an in-memory append only table.
+     *
+     * @param streamTable the input stream table
+     * @return an append-only in-memory table representing all data encountered in the stream
+     */
+    public static Table streamToAppendOnlyTable(Table streamTable) {
```

Comment:
We should give this a better name/and or path to use it.
Maybe make a `StreamTables` interface that offers this and other helpers? Or maybe put an instance method on `StreamToTableAdapter`?

Reply:
`StreamTools.toAppendOnlyTable`.

Reply:
`StreamTableTools.streamToAppendOnlyTable` because Python? Eww